pkg/providers/providers.go

```diff
@@ -20,34 +20,55 @@ type Provider interface {
 	IsAccessExpired(ctx context.Context, request *models.AccessRequest) (bool, error)
 }
 
+type ProviderKind string
+
+const (
+	ProviderKindMock       ProviderKind = "mock"
+	ProviderKindGitlab     ProviderKind = "gitlab"
+	ProviderKindGoogle     ProviderKind = "google"
+	ProviderKindTeleport   ProviderKind = "teleport"
+	ProviderKindAWS        ProviderKind = "aws"
+	ProviderKindCloudflare ProviderKind = "cloudflare"
```

Comment:
@JK-cto2b this addition in the list of provider will requires some adaptation in the UI side.